plugin/ScriptLoader/ScriptLoaderGenericConfigurationDefault.php

```diff
@@ -60,13 +66,199 @@ public function getScriptLoaderGenericConfiguration(): array
+		}
+
+		if ($this->isFormPage($pageContent) || ($this->isDetailEstatePage($pageContent) && ! empty($shortcodeFormForDetailPage))) {
+			$scripts = $this->renderScriptForFormPage($scripts, $pluginPath, $script);
+		}
+
+		return $scripts;
+	}
+
+	/**
+	 * @param string $content
+	 * @return bool
+	 */
+	private function isEstateListPage(string $content): bool
+	{
+		return $this->matchesShortcode($content, self::ESTATE_TAG, 'view', '[^"]*') &&
+				!$this->matchesShortcode($content, self::ESTATE_TAG, 'view', 'detail') ||
+				$this->matchesShortcode($content, self::ESTATE_TAG, 'units', '[^"]*');
+	}
+
+	/**
+	 * @param string $content
+	 * @return bool
+	 */
+	private function isDetailEstatePage(string $content): bool
+	{
+		return $this->matchesShortcode($content, self::ESTATE_TAG, 'view', 'detail');
+	}
+
+	/**
+	 * @param string $content
+	 * @return bool
+	 */
+	private function isFormPage(string $content): bool
+	{
+		return $this->matchesShortcode($content, self::FORM_TAG, 'form', '[^"]*');
+	}
+
+	/**
+	 * @param string $content
+	 * @param string $tag
+	 * @param string $attribute
+	 * @param string $valuePattern
+	 * @return bool
+	 */
+	private function matchesShortcode(string $content, string $tag, string $attribute, string $valuePattern): bool
+	{
+		$pattern = '/\[' . $tag . '\s+' . $attribute . '="' . $valuePattern . '"\]/';
+		return (bool) preg_match($pattern, $content);
+	}
+
+	/**
+	 * @param array $scripts
+	 * @param string $pluginPath
+	 * @param string $script
+	 * @param string $style
+	 * @param string $defer
+	 * @return array
+	 */
+
+	private function renderScriptForFormPage(array $scripts, string $pluginPath, string $script): array
+	{
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-custom-select', plugins_url('/dist/onoffice-custom-select.min.js', $pluginPath)))
+				->setDependencies(['jquery'])
+				->setLoadInFooter(true);
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-multiselect', plugins_url('/dist/onoffice-multiselect.min.js', $pluginPath)))
+				->setDependencies([])
+				->setLoadInFooter(true);
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-estatetype', plugins_url('/dist/onoffice-estatetype.min.js', $pluginPath)))
+				->setDependencies(['onoffice-multiselect'])
+				->setLoadInFooter(true);
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-leadform', plugins_url('/dist/onoffice-leadform.min.js', $pluginPath)))
+				->setDependencies(['jquery'])
+				->setLoadInFooter(true);
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-prevent-double-form-submission', plugins_url('/dist/onoffice-prevent-double-form-submission.min.js', $pluginPath)))
+				->setDependencies(['jquery'])
+				->setLoadInFooter(true);
+		$scripts[] = (new IncludeFileModel($script, 'onoffice-form-preview', plugins_url('/dist/onoffice-form-preview.min.js', $pluginPath)))
+				->setDependencies(['jquery'])
+				->setLoadInFooter(true);
+
+		if (get_option('onoffice-settings-honeypot') == true) {
```

Comment:
In the old code there was a check if the form is not an applicant search form. 
Is this still necessary?

Reply:
> In the old code there was a check if the form is not an applicant search form. Is this still necessary?

Yes, it can be a condition for loading JavaScript when necessary. I have added it